trappy/ftrace.py

```diff
@@ -306,14 +360,18 @@ def __populate_data(self, fin, cls_for_unique_word):
 
             if not self.basetime:
                 self.basetime = timestamp
-
-            if (timestamp < self.window[0] + self.basetime) or \
-               (timestamp < self.abs_window[0]):
+                # Disregard windows if updating/creating cache
+                # This way the cache is ready for later window requests
+                if not self.__class__.disable_cache:
```

Comment:
I think this is unecessary complexity: disabled caching is supposed to be a niche feature for testing and debugging, so we don't care about performance in that case. So I think when parsing a trace file we can _always_ ignore the window, and then _always_ apply the window after parsing is finished, using the new `windowify` methods. Does that work?

Reply:
That would work, but the incentive for doing this with a cache is that we're paying a cost (parsing time) that will later on be cushioned by the use of the cache. If caching is disabled, we should still mind the window so we stop parsing the trace once we're after the window.  I understand disabled cache should become an exotic use-case but I felt like this was still somewhat justified.

Reply:
I see where you're coming from but I think we should make our task as easy as possible by only having one use case that we really optimise, i.e. `cache_disabled=False`

Reply:
Okay, I'll remove that bit.